app/src/main/kotlin/io/intrepid/skotlinton/base/BasePresenter.kt

```diff
@@ -8,57 +11,31 @@ import io.reactivex.Observable
 import io.reactivex.Scheduler
 import io.reactivex.Single
 import io.reactivex.disposables.CompositeDisposable
+import timber.log.Timber
 
-abstract class BasePresenter<V : BaseContract.View>(protected var view: V?, configuration: PresenterConfiguration) : BaseContract.Presenter {
+abstract class BasePresenter<S : BaseScreen>(protected var screen: S, configuration: PresenterConfiguration) : LifecycleObserver {
 
-    protected val ioScheduler: Scheduler = configuration.ioScheduler
-    protected val uiScheduler: Scheduler = configuration.uiScheduler
+    private val ioScheduler: Scheduler = configuration.ioScheduler
+    private val uiScheduler: Scheduler = configuration.uiScheduler
```

Comment:
these should remain protected

Reply:
I can see this becoming a problem with `Observable.delay` since we probably want to pass in a scheduler for testing. This can be avoided with scheduler overriding and also exposing other methods for `applySchedulers`. What do you think?

Reply:
The schedulers are used for more than just applyschedulers. Sometimes the subclasses need to switch the schedulers manually